# Project 3 -- Zeru Zhou

**TA Help:** NA


    
**Collaboration:** NA
    
- Get help at piazza
- Get help from videos posted by Dr. Ward

## Question 1

In [1]:
list.files("/depot/datamine/data/olympics")

[1] "athlete_events.csv" "regions.csv"

In [3]:
dim(read.csv("/depot/datamine/data/olympics/athlete_events.csv"))

[1] 271116     15

In [4]:
dim(read.csv("/depot/datamine/data/olympics/regions.csv"))

[1] 230   3

In [1]:
olympics <- read.csv("/depot/datamine/data/olympics/athlete_events.csv")

In [8]:
dim(olympics)

[1] 271116     15

In [6]:
head(olympics)

ID Name                     Sex Age Height Weight Team           NOC
1 1  A Dijiang                M   24  180    80     China          CHN
2 2  A Lamusi                 M   23  170    60     China          CHN
3 3  Gunnar Nielsen Aaby      M   24   NA    NA     Denmark        DEN
4 4  Edgar Lindenau Aabye     M   34   NA    NA     Denmark/Sweden DEN
5 5  Christine Jacoba Aaftink F   21  185    82     Netherlands    NED
6 5  Christine Jacoba Aaftink F   21  185    82     Netherlands    NED
  Games       Year Season City      Sport        
1 1992 Summer 1992 Summer Barcelona Basketball   
2 2012 Summer 2012 Summer London    Judo         
3 1920 Summer 1920 Summer Antwerpen Football     
4 1900 Summer 1900 Summer Paris     Tug-Of-War   
5 1988 Winter 1988 Winter Calgary   Speed Skating
6 1988 Winter 1988 Winter Calgary   Speed Skating
  Event                              Medal
1 Basketball Men's Basketball        NA   
2 Judo Men's Extra-Lightweight       NA   
3 Football Men's Football            NA   
4 Tug-Of-War Men's Tug-Of-War        Gold 
5 Speed Skating Women's 500 metres   NA   
6 Speed Skating Women's 1,000 metres NA

In [7]:
str(olympics)

'data.frame':	271116 obs. of  15 variables:
 $ ID    : int  1 2 3 4 5 5 5 5 5 5 ...
 $ Name  : chr  "A Dijiang" "A Lamusi" "Gunnar Nielsen Aaby" "Edgar Lindenau Aabye" ...
 $ Sex   : chr  "M" "M" "M" "M" ...
 $ Age   : int  24 23 24 34 21 21 25 25 27 27 ...
 $ Height: int  180 170 NA NA 185 185 185 185 185 185 ...
 $ Weight: num  80 60 NA NA 82 82 82 82 82 82 ...
 $ Team  : chr  "China" "China" "Denmark" "Denmark/Sweden" ...
 $ NOC   : chr  "CHN" "CHN" "DEN" "DEN" ...
 $ Games : chr  "1992 Summer" "2012 Summer" "1920 Summer" "1900 Summer" ...
 $ Year  : int  1992 2012 1920 1900 1988 1988 1992 1992 1994 1994 ...
 $ Season: chr  "Summer" "Summer" "Summer" "Summer" ...
 $ City  : chr  "Barcelona" "London" "Antwerpen" "Paris" ...
 $ Sport : chr  "Basketball" "Judo" "Football" "Tug-Of-War" ...
 $ Event : chr  "Basketball Men's Basketball" "Judo Men's Extra-Lightweight" "Football Men's Football" "Tug-Of-War Men's Tug-Of-War" ...
 $ Medal : chr  NA NA NA "Gold" ...


Olympics has 271116 rows and 15 columns, and there are 3 types of data: integer, numeric and character. Each row contains basic information about a specific athlete.

## Question 2

In [10]:
length(unique(olympics$Sport))

[1] 66

In [13]:
unique(olympics$Sport)

[1] "Basketball"                "Judo"                     
 [3] "Football"                  "Tug-Of-War"               
 [5] "Speed Skating"             "Cross Country Skiing"     
 [7] "Athletics"                 "Ice Hockey"               
 [9] "Swimming"                  "Badminton"                
[11] "Sailing"                   "Biathlon"                 
[13] "Gymnastics"                "Art Competitions"         
[15] "Alpine Skiing"             "Handball"                 
[17] "Weightlifting"             "Wrestling"                
[19] "Luge"                      "Water Polo"               
[21] "Hockey"                    "Rowing"                   
[23] "Bobsleigh"                 "Fencing"                  
[25] "Equestrianism"             "Shooting"                 
[27] "Boxing"                    "Taekwondo"                
[29] "Cycling"                   "Diving"                   
[31] "Canoeing"                  "Tennis"                   
[33] "Modern Pentathlon"         "Figure Skating"           
[35] "Golf"                      "Softball"                 
[37] "Archery"                   "Volleyball"               
[39] "Synchronized Swimming"     "Table Tennis"             
[41] "Nordic Combined"           "Baseball"                 
[43] "Rhythmic Gymnastics"       "Freestyle Skiing"         
[45] "Rugby Sevens"              "Trampolining"             
[47] "Beach Volleyball"          "Triathlon"                
[49] "Ski Jumping"               "Curling"                  
[51] "Snowboarding"              "Rugby"                    
[53] "Short Track Speed Skating" "Skeleton"                 
[55] "Lacrosse"                  "Polo"                     
[57] "Cricket"                   "Racquets"                 
[59] "Motorboating"              "Military Ski Patrol"      
[61] "Croquet"                   "Jeu De Paume"             
[63] "Roque"                     "Alpinism"                 
[65] "Basque Pelota"             "Aeronautics"

There are 66 unique sports in the dataset olympics. I use 'unique' function to show unique individual sports included in the dataset. There are some sports I never expected like 'Motorboating' since I thought it is more like a hobby and just for fun. Never expected it is on olympics.

## Question 3

In [3]:
us_athletes <- subset(olympics, NOC=='USA')

In [4]:
dim(us_athletes)

[1] 18853    15

In [5]:
china_athletes <- subset(olympics, NOC=='CHN')

In [6]:
dim(china_athletes)

[1] 5141   15

In [7]:
both <- subset(olympics, NOC=='USA'| NOC=='CHN')

In [8]:
dim(both)

[1] 23994    15

There are 18853 rows in 'us_athletes' dataset; There are 5154 rows in dataset contains athletes from China named 'china_athletes'; There are 23994 rows in 'both' dataset.

## Question 4

In [9]:
prop.table(table(olympics$Sex[olympics$NOC=='USA']))


        F         M 
0.2934811 0.7065189 

In [8]:
prop.table(table(olympics$Sex[(olympics$NOC=='USA') & (olympics$Medal=='Gold')]))


        F         M 
0.3229719 0.6770281 

In [11]:
prop.table(table(olympics$Sex[olympics$NOC=='CHN']))


        F         M 
0.5388057 0.4611943 

In [9]:
prop.table(table(olympics$Sex[(olympics$NOC=='CHN') & (olympics$Medal=='Gold')]))


  F   M 
0.6 0.4 

29.35% athletes in the US are women; 32.30% athletes in the US with gold medals are women. 53.88% athletes in China are women; 60% athletes in China with gold medals are women.

## Question 5

In [2]:
us_athletes <- subset(olympics, NOC=='USA')

In [16]:
us_age <- us_athletes$Age

In [12]:
which.max(us_age)

[1] 17647

In [14]:
us_athletes[which.max(us_age), c('Age', 'Sport', 'Year')]

Age Sport            Year
257055 97  Art Competitions 1928

In [18]:
china_athletes <- subset(olympics, NOC=='CHN')

In [20]:
ch_age <- china_athletes$Age

In [21]:
which.max(ch_age)

[1] 1130

In [22]:
china_athletes[which.max(ch_age), c('Age', 'Sport', 'Year')]

Age Sport         Year
100160 45  Equestrianism 2008

For the US, the oldest athlete is 97 years old, the sport is 'Art Competitions', and the olympics year is 1928. For China, the oldest athlete is 45 years old, the sport is 'Equestrianism', and the olympics year is 2008.

## Pledge

By submitting this work I hereby pledge that this is my own, personal work. I've acknowledged in the designated place at the top of this file all sources that I used to complete said work, including but not limited to: online resources, books, and electronic communications. I've noted all collaboration with fellow students and/or TA's. I did not copy or plagiarize another's work.

> As a Boilermaker pursuing academic excellence, I pledge to be honest and true in all that I do. Accountable together – We are Purdue.